File: 2017 COMLAN/DetectRecursivity/src-weaved/matlab/foobarfoo.m
function foobarfoo()
   foo(1);
end

function [a] = foo(seed)
   a = 0;
   for i = seed:5
      a = a + bar(i);
   end
end

function [result] = bar(a)
   if (a > 0)
      result = a;
      return;
   end
   result = foo(a + 1);
end

File: 2017 COMLAN/DetectRecursivity/src/java/FooBarFoo.java
public class FooBarFoo {

    public double bar(int a) {
        if(a > 0) {
			return a;
		}
		
		return foo(a+1);
    }

    public double foo(int seed) {
        double a = 0;

        for (int i = seed; i < 5; i++) {
            a += bar(i);
        }

        return a;
    }

    public static void main(String[] args) {
        new FooBarFoo().foo(0);
    }
}
